src/app/component/ProductCustomizableOptions/ProductCustomizableOptions.container.js

```diff
@@ -0,0 +1,222 @@
+            || selectedDropdownOptions !== prevSelectedDropdownOptions
+        ) {
+            this.updateSelectedOptions();
+        }
+    }
+
+    stopLoading() {
+        this.setState({ isLoading: false });
+    }
+
+    async updateSelectedOptionsArray() {
+        const { getSelectedCustomizableOptions } = this.props;
+        const { selectedCheckboxValues } = this.state;
+        const customizable_options = [];
+
+        if (selectedCheckboxValues.length) {
+            selectedCheckboxValues.map(item => customizable_options.push(item));
+        }
+
+        getSelectedCustomizableOptions(customizable_options, true);
+    }
+
+    async updateSelectedOptions() {
+        const { getSelectedCustomizableOptions } = this.props;
+        const {
+            selectedDropdownOptions,
+            selectedFieldValue,
+            selectedAreaValue
+        } = this.state;
+        const customizable_options = [];
+
+        if (selectedFieldValue.length) {
+            selectedFieldValue.map(item => customizable_options.push(item));
+        }
+
+        if (selectedAreaValue.length) {
+            selectedAreaValue.map(item => customizable_options.push(item));
+        }
+
+        if (selectedDropdownOptions.length) {
+            selectedDropdownOptions.map(item => customizable_options.push(item));
+        }
+
+        getSelectedCustomizableOptions(customizable_options);
+    }
+
+    getRequiredOptionsData(options) {
+        const { getRequiredCustomizableOptions } = this.props;
+        const optionData = [];
+
+        options.map(({ option_id, required }) => {
+            if (required) {
+                return optionData.push({ option_id });
+            }
+
+            return null;
+        });
+
+        getRequiredCustomizableOptions(optionData);
+    }
+
+    setCustomizableOptionFieldValue(option_id, option_value) {
+        if (option_value) {
+            this.setState({ selectedFieldValue: [{ option_id, option_value }] });
+        } else {
+            this.setState({ selectedFieldValue: [] });
+        }
+    }
+
+    setCustomizableOptionAreaValue(option_id, option_value) {
+        if (option_value) {
+            this.setState({ selectedAreaValue: [{ option_id, option_value }] });
+        } else {
+            this.setState({ selectedAreaValue: [] });
+        }
+    }
+
+    setSelectedCheckboxValues(option_id, option_value) {
+        const { selectedCheckboxValues } = this.state;
+        const selectedValue = { option_id, option_value };
+
+        if (selectedCheckboxValues.some(item => option_value === item.option_value)) {
+            this.setState({
+                selectedCheckboxValues: selectedCheckboxValues.filter(value => value.option_value !== option_value)
+            });
+        } else {
+            this.setState({
+                selectedCheckboxValues: [...selectedCheckboxValues, selectedValue]
+            });
+        }
+    }
+
+    setSelectedDropdownValue(value = null, option_id) {
```

Comment:
This declaration makes no sense, `value = null` will never be fallback-ed to null.

Reply:
As the `option_id` is required argument.